flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/api/java/BatchTableEnvironment.java

```diff
@@ -0,0 +1,289 @@
+	 * Registers a {@link TableFunction} under a unique name in the TableEnvironment's catalog.
+	 * Registered functions can be referenced in Table API and SQL queries.
+	 *
+	 * @param name The name under which the function is registered.
+	 * @param tableFunction The TableFunction to register.
+	 * @param <T> The type of the output row.
+	 */
+	<T> void registerFunction(String name, TableFunction<T> tableFunction);
+
+	/**
+	 * Registers an {@link AggregateFunction} under a unique name in the TableEnvironment's catalog.
+	 * Registered functions can be referenced in Table API and SQL queries.
+	 *
+	 * @param name The name under which the function is registered.
+	 * @param aggregateFunction The AggregateFunction to register.
+	 * @param <T> The type of the output value.
+	 * @param <ACC> The type of aggregate accumulator.
+	 */
+	<T, ACC> void registerFunction(String name, AggregateFunction<T, ACC> aggregateFunction);
+
+	/**
+	 * Converts the given {@link DataSet} into a {@link Table}.
+	 *
+	 * <p>The field names of the {@link Table} are automatically derived from the type of the
+	 * {@link DataSet}.
+	 *
+	 * @param dataSet The {@link DataSet} to be converted.
+	 * @tparam T The type of the {@link DataSet}.
+	 * @return The converted {@link Table}.
+	 */
+	<T> Table fromDataSet(DataSet<T> dataSet);
+
+	/**
+	 * Converts the given {@link DataSet} into a {@link Table} with specified field names.
+	 *
+	 * Example:
+	 *
+	 * {{{
+	 *   DataSet<Tuple2<String, Long>> set = ...
+	 *   Table tab = tableEnv.fromDataSet(set, "a, b");
+	 * }}}
+	 *
+	 * @param dataSet The {@link DataSet} to be converted.
+	 * @param fields The field names of the resulting {@link Table}.
+	 * @tparam T The type of the {@link DataSet}.
+	 * @return The converted {@link Table}.
+	 */
+	<T> Table fromDataSet(DataSet<T> dataSet, String fields);
+
+	/**
+	 * Registers the given {@link DataSet} as table in the
+	 * {@link TableEnvironment}'s catalog.
+	 * Registered tables can be referenced in SQL queries.
+	 *
+	 * The field names of the {@link Table} are automatically derived from the type of the{@link DataSet}.
+	 *
+	 * @param name The name under which the {@link DataSet} is registered in the catalog.
+	 * @param dataSet The {@link DataSet} to register.
+	 * @tparam T The type of the {@link DataSet} to register.
+	 */
+	<T> void registerDataSet(String name, DataSet<T> dataSet);
+
+	/**
+	 * Registers the given {@link DataSet} as table with specified field names in the
+	 * {@link TableEnvironment}'s catalog.
+	 * Registered tables can be referenced in SQL queries.
+	 *
+	 * Example:
+	 *
+	 * {{{
+	 *   DataSet<Tuple2<String, Long>> set = ...
+	 *   tableEnv.registerDataSet("myTable", set, "a, b");
+	 * }}}
+	 *
+	 * @param name The name under which the {@link DataSet} is registered in the catalog.
+	 * @param dataSet The {@link DataSet} to register.
+	 * @param fields The field names of the registered table.
+	 * @tparam T The type of the {@link DataSet} to register.
```

Comment:
Convert to `@param <T>`.

Could you please check other javadocs ?

Reply:
I will grep and replace them once for all. Sorry for this.